### DISCUSSION TITLE
SSL with and without server name

### DISCUSSION THREAD

**Discussion by epsilon-0** [uNetworking/uWebSockets#1101] [Q&A] [answered]
I've been using `uWS::TemplateApp<false>` and it's been working awesomely.
So I decided to step it up to using SSL and its been *running* so far, but not getting any inputs or connections.

Basically, I've followed the example in the [HttpServer](https://github.com/uNetworking/uWebSockets/blob/master/examples/HttpServer.cpp) and my server runs but takes no input, from POST, while the non-ssl version takes the POST input.
I was wondering if I need to do something like in the [ServerName](https://github.com/uNetworking/uWebSockets/blob/master/examples/ServerName.cpp) example if I want to accept POST requests?

Just thought it would be better to document this in a small issue.

Thanks!
Aisha

**Comment by ghost** [comment, marked as answer]
There are companies running with SSL in production so I'm sure it works. You don't need to add a server name.

**Comment by epsilon-0** [comment]
True, I wasn't implying it doesn't work, just wanted to know if I was missing something obvious.

> You don't need to add a server name.

Thanks for telling this. Knowing this helps in debugging.

**Comment by fuxey** [comment]
Hi everyone! 

I am also struggling with the SSL connection. 
so I am experimenting with the EchoServer example and tied it to my application. 
uWS::TemplateApp<false> works perfectly fine.

when I compile with SSL support plus add the key stuff 
```
	openssl genrsa 4096 > `key.pem`
	openssl req -new -key key.pem | openssl x509 -days 365 -req -signkey key.pem > cert.pem 
```

then the server starts listening, but I am unable to connect via websocket neither ws://localhost:9001 nor wss://localhost:9001.
I am also wondering if there's an option where to select between tls1.3 and tls.1.2. 
Thanks for any help!

**Reply by FabrizioCheloni** [reply]
Having the same issue here, did you ever found a solution?
